Fix parsing of MP4 and FLAC files, add tests and some groundwork for Ogg file parsing

Holzhaus opened the pull request.

uklotzde commented on src/track/serato/markers.cpp:
```diff
@@ -182,8 +200,87 @@ SeratoMarkersEntryPointer SeratoMarkersEntry::parse(const QByteArray& data) {
+                    color,
+                    type,
+                    isLocked));
+    qDebug() << "SeratoMarkersEntry" << *pEntry;
```
We should reduce this log spam by logging conditionally. Otherwise all debug logs appear in the log file.

Holzhaus commented:
Ok, I probably should stop pushing more commits here. This becomes very hard to review.
I think this already is in a mergeable state:

- Added a new parser for the "Markers_" tag in MP4 files (slightly different format than in ID3 tags)
- Added tests for MP4
- Added a way to use a different code path for Ogg and FLAC Xiph tags (different format, different tag name)
- Add some roundtrip tests for Markers2 data in FLAC (parsing works fine, but roundtripping data with hotcues does not work yet because the last base64 padding character looks different sometimes, so I only added tests without hotcues for now)
- Added Markers2 parsing for Ogg file (but it's disabled for now, since there are other bits missing)
- Added roundtrip tests for Markers2 data in Ogg files

Still missing:
- Fix FLAC Markers2 padding to allow lossless roundtrips and more tests
- Parser for Ogg Markers tag
- Fusing of Ogg tag data (the Markers and Markers2 tags are incomplete without the other)

I think these can also go into a separate PR though.
